### PR TITLE
Add PhpVersion param on isSmallerThanOrEqual and isGreaterThanOrEqual

### PR DESCRIPTION
**Pull request by VincentLanglet**
As recommended in https://github.com/phpstan/phpstan/issues/11732#issuecomment-2371803227

I'll do the implementation/fix in another PR, to be sure to not delay the signature update.

### CODE REVIEW

File: src/Type/CompoundType.php [hunk 1]
```diff
@@ -16,6 +17,6 @@ public function isAcceptedWithReasonBy(Type $acceptingType, bool $strictTypes):
 
 	public function isGreaterThan(Type $otherType): TrinaryLogic;
```
Comment (ondrejmirtes) [inline src/Type/CompoundType.php]: What about here?

File: src/Type/Type.php [hunk 2]
```diff
@@ -217,7 +217,7 @@ public function toArrayKey(): Type;
 
 	public function isSmallerThan(Type $otherType): TrinaryLogic;
```
Comment (ondrejmirtes) [inline src/Type/Type.php]: What about here?

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]
Also other potential methods to add it to:

* getSmallerType
* getSmallerOrEqualType
* getGreaterType
* getGreaterOrEqualType

File: src/Type/CompoundType.php (hunk 1 above)
Comment (VincentLanglet) [reply src/Type/CompoundType.php]: Indeed, `var_dump('' < 0);` is true on PHP 8 but false on PHP 7.4

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]
Mention it in UPGRADING.md in minor breaks section at the bottom. Thanks!

**Comment by staabm** [comment]
I think we need a similar PR for methods which currently use `PhpVersionStaticAccessor::getInstance()` instead of using a `PhpVersion` parameter explicitly?

**Comment by ondrejmirtes** [comment]
That would be awesome!

**Comment by VincentLanglet** [comment]
> I think we need a similar PR for methods which currently use `PhpVersionStaticAccessor::getInstance()` instead of using a `PhpVersion` parameter explicitly?

It's used by 
- StubValidator => seems easy to change
- `ConstantArrayType::findTypeAndMethodNames` => Need to pass `phpVersion` to `isCallable` and `getCallableParametersAcceptors`
- `ConstantStringType::isCallable`

Changing the signature of `isCallable` seems to have a bigger impact since it's used in
- `Type::inferTemplateTypes`
- `Type::isSubTypeOf`
- `Type::isSuperTypeOf`
- `Type::acceptsWithReason`
- `Type::traverseSimultaneously`

And changing those signature lead to too many changes...

**Comment by ondrejmirtes** [comment]
Yeah, I wouldn't want to do this for `Type::isCallable()` / `ConstantArrayType::findTypeAndMethodNames`. It's just covering a quirk/edge-case of pre-PHP 8.0 behaviour.

**Comment by ondrejmirtes** [comment]
Thank you.

**Pull request merged**